src/data/list/dedup.lean

```diff
@@ -77,4 +77,42 @@ lemma repeat_dedup {x : α} : ∀ {k}, k ≠ 0 → (repeat x k).dedup = [x]
 | (n+2) _ := by rw [repeat_succ, dedup_cons_of_mem (mem_repeat.2 ⟨n.succ_ne_zero, rfl⟩),
     repeat_dedup n.succ_ne_zero]
 
+lemma count_dedup (l : list α) (a : α) :
+  l.dedup.count a = if a ∈ l then 1 else 0 :=
+begin
+  refine trans (count_eq_of_nodup $ nodup_dedup l) _,
+  simp_rw [mem_dedup],
+end
```

Comment:
Using `refine trans` like this isn't that common. In this case you can write
```suggestion
by simp_rw [count_eq_of_nodup (nodup_dedup l), mem_dedup]
```
or
```suggestion
(count_eq_of_nodup $ nodup_dedup l).trans $ by simp_rw [mem_dedup]
```
if you really like `trans`